content/reference/limits.markdown

```diff
@@ -0,0 +1,76 @@
+---
+layout: default
+title: Limits
+categories: [Reference, Limits]
+---
+
+There are various limits enforced in CFEngine, for names and values in both policy, web UI, and other places.
+It can sometimes be relevant for both developers and users to know what limits are in place.
+The term limits is used broadly, referring to different kinds of limits (character limits, rate limits, allowed / blocked characters, etc.).
+
+## Limits in policy language and evaluation
+
+**Variable names:** Maximum 1024 characters.
+
+**Variable values (strings):** No explicit maximum, we've tested strings of several megabytes, however this is not recommended for several reasons:
+
+- When strings get excessively long, everything you do with them takes a long time.
+  To test this and give some kind of idea: parsing and storing a ~3MB string in a variable took 25 seconds on the author's laptop.
+- If you want to include these variables in reporting data, they will be truncated.
+  (See below for details).
+- Any time you print such a variable, there will be a lot of output.
+  (And printing this data to file or console is also slow).
+- `cf-agent` will exit if you create variables so large that you run out of memory.
+- Historically, some built-in functions in policy language have truncated strings (usually to around 4K bytes).
+  This is not intended / desirable, we are fixing those, but it might be relevant for you, especially if you use older versions of CFEngine.
```

Comment:
```suggestion
  This is not intended / desirable; we are fixing those, but it might be relevant for you, especially if you use older versions of CFEngine.
```